zerver/lib/upload/__init__.py

```diff
@@ -114,20 +114,27 @@ def get_public_upload_root_url() -> str:
     return upload_backend.get_public_upload_root_url()
 
 
-def sanitize_name(value: str) -> str:
-    """
-    Sanitizes a value to be safe to store in a Linux filesystem, in
+def sanitize_name(value: str, *, strict: bool = False) -> str:
+    """Sanitizes a value to be safe to store in a Linux filesystem, in
     S3, and in a URL.  So Unicode is allowed, but not special
     characters other than ".", "-", and "_".
 
+    In "strict" mode, it does not allow Unicode, allowing only ASCII
+    [A-Za-z0-9_] as word characters.  This is for the benefit of tusd,
+    which is not Unicode-aware.
```

Comment:
The commin name says `tusd: Remove non-Unicode characters from path-ids. ` - was it supposed to say non-ASCII?

Reply:
Ah, very much yes